src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Workspace/Mef/MefWorkspaceServices.cs

```diff
@@ -51,7 +52,18 @@ public override HostServices HostServices
 
         internal IMefHostExportProvider HostExportProvider => _exportProvider;
 
-        public override Workspace Workspace => _workspace;
+        internal string WorkspaceKind => _workspace.Kind;
+
+        public override Workspace Workspace
+        {
+            get
+            {
+                //#if !CODE_STYLE
+                //                Contract.ThrowIfTrue(_workspace.Kind == CodeAnalysis.WorkspaceKind.RemoteWorkspace, "Access .Workspace off of a RemoteWorkspace MefWorkspaceServices is not supported.");
+                //#endif
```

Comment:
woudl like to enable this, but it fires too many times currently.  WIP on this part :)